mitodl/mit-learn
# v2 drawer certification updates

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5976

### Description (What does it do?)
This PR changes the way prices and certification are displayed in the new learning resource drawer. Free courses that offer a certificate still display said certification price next to the "Free" text in the price info item. Paid courses with certification now display a separate info item with the certification type, based on the `certification_type` property of the resource.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/185511ea-e89a-4952-a349-acec07f13ce4)
![image](https://github.com/user-attachments/assets/82e8fbed-1c47-454c-9cd5-b30020e06294)
![image](https://github.com/user-attachments/assets/6ed677ef-f673-465b-8b10-6badf2906bca)
![image](https://github.com/user-attachments/assets/17e98d9c-da32-442c-bba2-2c13adb19077)

### How can this be tested?
 - If you have Posthog set up locally, enable the `lr_drawer_v2` flag
 - If you don't have Posthog set up locally, you can force `drawerV2` to be `true` in `LearningResourceDrawer.tsx`
 - Spin up this branch of `mit-learn`
 - Ensure you have sufficient data backpopulated into your local instance to cover the following test cases:
   - Free course without certification
   - Free course with certification and no prices
   - Free course with certification and prices
   - Paid course with certification
 - Go to the search page at http://localhost:8062/search
 - In the "Certificate" search facet, check one of the boxes at a time and then examine the drawer for the courses presented in the search results:
   - "No Certificate" should display no certificate information whatsoever
   - "Professional Certificate" contains mostly paid courses with a certification, so you should see the price and certificate type each on their own line
   - "Certificate of Completion" contains mostly free courses that offer a paid certificate, so you should see the old behavior with "Price: Free" alongside the certificate price pill
   - "MicroMasters Credential" has a little bit of both behaviors described above; free courses with an optional paid certificate and paid courses with certificates
